packages/wrangler/src/index.tsx

```diff
@@ -677,7 +678,34 @@ function createCLIParser(argv: string[]) {
 				sendMetrics: config.send_metrics,
 			});
 
-			const scriptName = getLegacyScriptName(args, config);
+			let scriptName;
+
+			// Worker names can't contain "." (and most routes should), so use that as a discriminator
+			if (args.name?.includes(".")) {
+				const zone = await getZoneForRoute(args.name);
+				if (!zone) {
+					throw new Error(`Could not find zone for route ${args.name}`);
```

Comment:
This error message should be more actionable by the user -- describe what exactly went wrong and how they can fix it.